### PR TITLE
Proposition change template ES6

### PR DESCRIPTION
**Pull request by Peltoche**
- Add babel config in client brunch config
- Move from old module system to the new one
- Move class attributes in contructor (specs need)
- Set some getters / setters to be Backbone and ES6 compliant

### CODE REVIEW

**Comment by nono** [comment]
@jsilvestre what do you think?

File: client/app/views/app_view.es6 [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+   */
+    this.$el = $('body.application');
+
+    this.model = {};
+
+    this.template = template;
   }
 
-  afterRender() {
+  afterRender () {
```
Comment (jsilvestre) [inline client/app/views/app_view.es6]: Why is there a space here?

File: client/app/application.es6 [hunk 2]
```diff
@@ -13,5 +12,5 @@ module.exports = {
     if (typeof Object.freeze === 'function') {
       Object.freeze(this);
     }
-  },
-};
+  }
+}
```
Comment (jsilvestre) [inline client/app/application.es6:15]: I am not sure this deserved a class. It's just a function.

**Comment by jsilvestre** [comment]
I would recommend against using classes when they are not necessary, since it's a bit of a controversial feature after all. No strong feeling though, your call @nono.

**Comment by nono** [comment]
It's OK for me. It's a bit overkill to use a class for a simple case like this, but it shows how to do it for more complex cases (like several other features of the template).

File: client/app/lib/base_view.es6 [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-  template() {}
+export default class BaseView extends Backbone.View {
+
+  template() {
+    return '';
+  }
 
   getRenderData() {
-    return { model: this.model.toJSON() };
+    return { model: JSON.stringify(this.model) };
```
Comment (nono) [inline client/app/lib/base_view.es6]: @Peltoche you are sure about this change?
Backbone models saves its attributes in an hidden field and use getter/setter to access them. If you do a JSON.stringify, I think you'll have all the things nested in an `attributes` object. Correct me if I am wrong.

File: client/app/lib/view_collection.es6 [hunk 4]
```diff
@@ -14,10 +14,18 @@ const BaseView = require('lib/base_view');
...
-  itemview: {}
+export default class ViewCollection extends BaseView {
+  // bind listeners to the collection
+  constructor() {
+    super();
+    this.itemview = {};
 
-  views: {}
+    this.views = {};
+    this.views = {};
```
Comment (nono) [inline client/app/lib/view_collection.es6]: @Peltoche it's two times the same line!

**Comment by nono** [comment]
@Peltoche did you look at my comments?

File: client/app/lib/base_view.es6 (hunk 3 above)
Comment (Peltoche) [inline client/app/lib/base_view.es6]: Oups, that's right... After consieration it seem not be the best idea to automatically send the object data so I will follow the JS template and return null by default.

**Pull request merged**